lib/server.go

```diff
@@ -1930,12 +1930,19 @@ func (srv *Server) _addNewTxn(
 		return nil, err
 	}
 
-	if srv.blockchain.chainState() != SyncStateFullyCurrent {
+	srv.blockchain.ChainLock.RLock()
+	tipHeight := uint64(srv.blockchain.BlockTip().Height)
+	srv.blockchain.ChainLock.RUnlock()
 
-		err := fmt.Errorf("Server._addNewTxnAndRelay: Cannot process txn "+
-			"from peer %v while syncing: %v %v", pp, srv.blockchain.chainState(), txn.Hash())
-		glog.Error(err)
-		return nil, err
+	if srv.blockchain.chainState() != SyncStateFullyCurrent {
+		// We allow txn relay if chain is in a need blocks state and is running PoS
+		if !(srv.blockchain.chainState() == SyncStateNeedBlocksss &&
+			srv.blockchain.params.IsPoSBlockHeight(tipHeight)) {
```

Comment:
The logic seems right but the negation on the whole condition makes it a bit harder to interpret. This should work just as well
```suggestion
		if chainState != SyncStateNeedBlocksss || !srv.blockchain.params.IsPoSBlockHeight(tipHeight)) {
```